libpromises/evalfunction.c

```diff
@@ -7388,7 +7388,7 @@ void ModuleProtocol(EvalContext *ctx, char *command, const char *line, int print
             else if (StringMatchFullWithPrecompiledRegex(context_name_rx, content))
             {
                 Log(LOG_LEVEL_VERBOSE, "Module changed variable context from '%s' to '%s'", context, content);
-                strcpy(context, content);
+                strlcpy(context, content, context_size);
```

Comment:
No, this can't overflow and hides logic. Keep the `strcpy` and just use `assert` or `ProgrammingError` for checking that `context_size > 50` preferably at the beginning of the file. Or at least I think, I didn't go through the whole function.